scipy/optimize/meson.build

```diff
@@ -206,6 +206,10 @@ endif
 
 _dummy_init_optimize = fs.copyfile('__init__.py')
 
+_cython_tree = [
```

Comment:
Ah, I wanted to push a code comment, but then realized this actually removes an existing dependency by overwriting the value of `_cython_tree`. I'll push a fix.

Reply:
My apologies. I have 0 experience from meson, hence it's easy to do something stupid for me.

Reply:
No worries at all. It's not obvious at first - the Meson DSL only has immutable variables but you can reassign variable names. So if `_cython_tree` was used in another `meson.build` file that's executed first, it gets reassigned/overwritten here and then future usages change.

Reply:
This still seems happy, so I'll merge.